### PR TITLE
More util

### PR DESCRIPTION
**Pull request by LightRobin**
adds a few util functions, undecided on whether steamID To/From 64(id) should be in game lib or bit lib

### CODE REVIEW

**Comment by Vurv78** [comment]
SharedRandom should be lower camel case and isn't getUserGroups a little intrusive? I also don't really see a use for it alongside sharedRandom, maybe that's just me

**Comment by LightRobin** [comment]
ah ye sharedrandom should be lower case, good spot, i dont think getusergroups is intrusive, on most all servers you can already see everyones usergroup on the leaderboard

**Comment by LightRobin** [comment]
might be better to add Player:GetUserGroup() instead, ill wait and see what sparky says

**Comment by Vurv78** [comment]
Oh, I didn't know we didn't have getUserGroup, yeah that's better imo. Why would you need to know the usergroups of players that aren't online anyway was my point, since the list of usergroups also keeps track of settings/users.txt

**Comment by thegrb93** [comment]
> 
> 
> SharedRandom should be lower camel case and isn't getUserGroups a little intrusive? I also don't really see a use for it alongside sharedRandom, maybe that's just me

getUserGroups isn't needed, you can just use getNWVar("Usergroup")

**Comment by thegrb93** [comment]
I think it just returns the default gmod admin list too. user.cfg or whatever it was.

File: lua/starfall/libs_sh/builtins.lua [hunk 1]
```diff
@@ -177,6 +177,12 @@ builtins_library.SERVER = SERVER
 -- @return Boolean
 builtins_library.isFirstTimePredicted = IsFirstTimePredicted
 
+--- Returns the current date formatted like '2012-10-31 18-00-00'
+-- @name builtins_library.dateStamp
+-- @class function
+-- @return string date
+builtins_library.dateStamp = util.DateStamp
```
Comment (thegrb93) [inline lua/starfall/libs_sh/builtins.lua:184]: This is just `os.date("%c")`

**Comment by thegrb93** [comment]
Should be good if we get rid of getUserGroups and DateStamp.

**Pull request merged**